ethereum/eth/src/test/java/org/hyperledger/besu/ethereum/eth/manager/task/GetHeadersFromPeerByHashTaskTest.java

```diff
@@ -97,9 +108,70 @@ public void getHeadersFromHashWithSkipReversed() {
     getHeadersFromHash(2, true);
   }
 
+  @Test
+  public void completesWhenPeersSendEmptyResponsesAndReducesReputation() {
+    final Responder responder = RespondingEthPeer.emptyResponder();
+    final RespondingEthPeer respondingEthPeer =
+        EthProtocolManagerTestUtil.createPeer(ethProtocolManager, 1000);
+
+    // Setup data to be requested
+    final List<BlockHeader> requestedData = generateDataToBeRequested();
+
+    // Execute task and wait for response
+    final EthTask<AbstractPeerTask.PeerTaskResult<List<BlockHeader>>> task =
+        createTask(requestedData);
+    final CompletableFuture<AbstractPeerTask.PeerTaskResult<List<BlockHeader>>> future = task.run();
+    respondingEthPeer.respond(responder);
+    assertThat(future.isDone()).isTrue();
+    assertThat(future.isCompletedExceptionally()).isFalse();
+    assertThat(respondingEthPeer.getEthPeer().getReputation().getScore()).isEqualTo(99);
+  }
+
+  @Test
+  public void completesWhenPeerSendsTooManyHeadersAndReducesReputation() {
+    // Setup a peer returning too many headers
+    final RespondingEthPeer respondingEthPeer =
+        EthProtocolManagerTestUtil.createPeer(ethProtocolManager, 1000);
+
+    // Setup data to be requested
+    final List<BlockHeader> requestedData =
+        generateDataToBeRequested(); // request DEFAULT_COUNT headers
+
+    // Execute task and wait for response
+    final EthTask<AbstractPeerTask.PeerTaskResult<List<BlockHeader>>> task =
+        createTask(requestedData);
+    final CompletableFuture<AbstractPeerTask.PeerTaskResult<List<BlockHeader>>> future = task.run();
+    respondingEthPeer.respond(
+        (c, m) -> Optional.of(BlockHeadersMessage.create(dummyBHList(DEFAULT_COUNT + 1))));
```

Comment:
DEFAULT_COUNT + 1 is what makes it "too many", right? 
can you move the related comment here